content/articles/working-with-java-synchronization/index.md

```diff
@@ -0,0 +1,226 @@
+        }
+      }).start();
+    }
+  }
+}
+```
+
+```
+The output of the thread is : Thread-0 - 3
+The output of the thread is : Thread-1 - 3
+The output of the thread is : Thread-2 - 3
+The output of the thread is : Thread-4 - 5
+The output of the thread is : Thread-3 - 4
+```
+
+As shown above, the threads are chosen at random, and the value is incorrect. The value should rise by 1 but that is not the case. Usually, the output is 3. Threads 0, 1, and 2 share the same value thus showing a race condition. After understanding what a race condition is, let's now look at how to refrain from it.
+
+It is evident that the crucial element (code that changes shared resources) must be limited. Java's synchronized keyword synchronizes shared resource access. This prevents thread interference during atomic operations.
+
+Synchronizing the method call should avoid the race problem:
+
+```Java
+public class Example2 {
+  int check = 1;
+  public  void CheckBoost(){
+    try {
+      Thread.sleep(1000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+    check++;
+  }
+  public int obtainCheck(){
+    return check;
+  }
+  public static void main(String[] args) {
+    Example2 zy = new Example2();
+    int x;
+    for(x = 1; x < 6; x++){
+      new Thread(new Runnable() {			
+        @Override
+        public void run() {
+          synchronized(zy){
+            zy.CheckBoost();
+            System.out.println("The output of the thread is : " + Thread.currentThread().getName() + " - " + zy.obtainCheck());
+          }
+        }
+      }).start();
+    }	
+  }
+}
+
+```
+
+```
+The output of the thread is : Thread-0 - 2
+The output of the thread is : Thread-4 - 3
+The output of the thread is : Thread-3 - 4
+The output of the thread is : Thread-2 - 5
+The output of the thread is : Thread-1 - 6
+```
+
+From the results of the threads above, no thread shares a value. Our aim of avoiding a race condition is fulfilled.
+
+### Understanding synchronized methods and synchronized blocks
+To understand the two, let us first look at each separately:
+
+#### Synchronized methods
+These consist of the following properties:
+
+- A synchronized method locks down the whole object. No other thread may access synchronized methods in the object while the method is executing. In non-static methods, this locks them whereas static methods are locked by their class.
+- Synchronized methods need locks on the current object or, in the case of static methods, the whole class. This is because the lock is obtained when the thread enters and released when the thread quits (naturally or by throwing an Exception).
+- Synchronized methods keep the lock for the method scope.
+- A synchronized static function prevents the class object from being changed. A class is a blueprint or template from which objects are built. As a result, an object is a class's instance(result).

```

Comment:
Here, when I said, *"Not sure what you mean by class object"* I didn't mean that you should explain the concept of classes and objects (the reader should already understand that given the skill level of this tutorial). I just felt that the wording was a little off when you said *class object*. I presume you meant **instances of the class.**

Reply:
I'll change this accordingly. Let me know if I'm wrong.

Reply:
I think it sounds great.